python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py

```diff
@@ -764,7 +796,15 @@ def _attach_metadata(
         """
 
         def _map_fn(event: DbtDagsterEventType) -> DbtDagsterEventType:
-            result = fn(self._dbt_cli_invocation, event)
+            with disable_dagster_warnings():
+                ctx = DbtAttachMetadataContext(
+                    event=event,
+                    invocation=self._dbt_cli_invocation,
+                    dbt_resource_props=_get_dbt_resource_props_from_event(
+                        self._dbt_cli_invocation, event
+                    ),
+                )
```

Comment:
It would be perfect to get the following:

1. The `dagster_event`
2. The `DbtCliEventMessage` that produced the `dagster_event`
3. The invocation
4. The `dbt_resource_props` (which can essentially be derived from (3), but just provided for ease of access to the user

That way, we can potentially deprecate use of `stream_raw_events()` in favor of these chaining methods on `DbtEventIterator`